components/src/main/java/uk/gov/hmrc/components/organism/menu/MenuPanelRowView.kt

```diff
@@ -0,0 +1,138 @@
+        binding.layout.apply {
+            contentDescription = when (countToDisplay) {
+                0 -> {
+                    context.getString(R.string.menu_panel_button_circle_notification_content_description,
+                            title,
+                            body)
+                }
+                1 -> {
+                    context.getString(R.string.menu_panel_button_one_notification_content_description,
+                            title,
+                            countToDisplay.toString(),
+                            body)
+                }
+                else -> {
+                    val quantity = if (countToDisplay in 2..MAXIMUM_NOTIFICATION_COUNT) {
+                        countToDisplay.toString()
+                    } else {
+                        context.getString(R.string.menu_panel_button_over_ninety_nine_notifications_text)
+                    }
+                    context.getString(R.string.menu_panel_button_multiple_notifications_content_description,
+                            title,
+                            quantity,
+                            body)
+                }
+            }
+        }
+    }
+
+    fun setTitle(title: CharSequence) {
+        binding.textTitle.text = title
+    }
+
+    fun getTitle(): CharSequence {
+        return binding.textTitle.text
+    }
+
+    fun setBody(body: CharSequence) {
+        binding.textBody.text = body
+    }
+
+    fun getBody(): CharSequence {
+        return binding.textBody.text
+    }
+
+    fun showNotification(notificationCount: Int = 0) {
+        binding.notification.apply {
+            if (notificationCount > 0) {
+                text = if (notificationCount > MAXIMUM_NOTIFICATION_COUNT) {
+                    context.getString(R.string.menu_panel_button_over_ninety_nine_notifications_text)
+                } else {
+                    notificationCount.toString()
+                }
+                background = context.getDrawable(R.drawable.components_menu_panel_round_square_notification)
+                setNotificationContentDescription(notificationCount)
+            } else {
+                background = context.getDrawable(R.drawable.components_menu_pane_circle_notification)
+                layoutParams.width = 0
+                setNotificationContentDescription(0)
+            }
+            visibility = View.VISIBLE
+        }
+    }
+
+    fun hideNotification() {
+        binding.notification.visibility = View.GONE
+        setDefaultContentDescription(getTitle(), getBody())
+    }
+
+    override fun removeChildPadding() {
+        val defaultPadding = resources.getDimensionPixelSize(R.dimen.hmrc_spacing_24)
+        val leftRightPadding = resources.getDimensionPixelSize(R.dimen.hmrc_spacing_16)
+
+        binding.textTitle.setMargins(leftRightPadding, defaultPadding, leftRightPadding, defaultPadding)
+        binding.textBody.setMargins(leftRightPadding, 0, leftRightPadding, defaultPadding)
+    }
```

Comment:
I don't think we need to have this logic here. We can just set the margins we need in the xml. This class does not really need to extend `PaddedComponent`. 

We use "PaddedComponent" for components we have built to allow children to be added when used (eg. adding a button or some extra text to a PrimaryCardView.). Different components have different padding rules, so eg. if we added a PrimaryButton to the PrimaryCardView, we would want to keep the Card component's padding, but if we added a SecondaryButton we would want to remove the child padding so that the touch state would be flush with the edges of the card.

We are not allowing children to be added to this component, so we do not need this logic. :)